src/lib/utils.js

```diff
@@ -177,75 +177,52 @@ export function deepCopy(x) {
-  let fill = activeLayersData.filter(d => d['type'] === 'fill' && !d['detail-view']);
-  if (fill.length) {
-    Object.keys(fill).forEach(key => {
-      if (map.getLayer(fill[key].id)) {
-        map.moveLayer(fill[key].id);
-      }
-    });
-    if (fill.find(c => c.id === nextLayerId) && map.getLayer(nextLayerId)) {
-      map.moveLayer(nextLayerId);
+    /**
+     * Rearranges rendered layers and puts selected layer on top. 
+     * @param {Object} filteredLayers - Mapbox layers
+     */
+    const moveLayers = filteredLayers => {
+        Object.keys(filteredLayers).forEach((key) => {
+            if (map.getLayer(filteredLayers[key].id)) {
+                map.moveLayer(filteredLayers[key].id);
+            }
+        });
+        if (
+            filteredLayers.find(d => d.id === nextLayerId) &&
+            map.getLayer(nextLayerId)
+        ) {
+            map.moveLayer(nextLayerId);
+        }
     }
-  }
 
-  let circles = activeLayersData.filter(d => d['type'] === 'circle');
-  if (circles.length) {
-    Object.keys(circles).forEach(key => {
-      if (map.getLayer(circles[key].id)) {
-        map.moveLayer(circles[key].id);
-      }
-    });
-    if (circles.find(c => c.id === nextLayerId) && map.getLayer(nextLayerId)) {
-      map.moveLayer(nextLayerId);
+
+    const fill = activeLayersData.filter(d => d['type'] === 'fill' && !d['detail-view']);
+    if (fill.length) {
+        moveLayers(fill);
     }
-  }
 
-  let symbols = activeLayersData.filter(d => d['type'] === 'symbol');
-  if (symbols.length) {
-    Object.keys(symbols).forEach(key => {
-      if (map.getLayer(symbols[key].id)) {
-        map.moveLayer(symbols[key].id);
-      }
-    });
-    if (symbols.find(s => s.id === nextLayerId) && map.getLayer(nextLayerId)) {
-      map.moveLayer(nextLayerId);
+    const circles = activeLayersData.filter(d => d['type'] === 'circle');
+    if (circles.length) {
+        moveLayers(circle);
     }
-  }
 
-  let line = activeLayersData.filter(d => d['type'] === 'line');
-  if (line.length) {
-    Object.keys(line).forEach(key => {
-      if (map.getLayer(line[key].id)) {
-        map.moveLayer(line[key].id);
-      }
-    });
-    if (line.find(c => c.id === nextLayerId) && map.getLayer(nextLayerId)) {
-      map.moveLayer(nextLayerId);
+    const symbols = activeLayersData.filter(d => d['type'] === 'symbol');
+    if (symbols.length) {
+        moveLayers(symbols);
     }
-  }
 
-  let detailViewActive = activeLayersData.filter(d => d['detail-view']);
-  if (detailViewActive.length) {
-    Object.keys(detailViewActive).forEach(key => {
-      if (map.getLayer(detailViewActive[key].id)) {
-        map.moveLayer(detailViewActive[key].id);
-      }
-    });
-    if (detailViewActive.find(d => d.id === nextLayerId) && map.getLayer(nextLayerId)) {
-      map.moveLayer(nextLayerId);
+    const line = activeLayersData.filter(d => d['type'] === 'line');
+    if (line.length) {
+        moveLayers(line);
```

Comment:
Should line layers be the second filtered layers which are moved? So that the other (non-fill) layers render on top of line layers?

Reply:
Was following the previous order but it makes more sense to have it at second place. Have made the change.

Reply:
> Was following the previous order

Cool totally, yeah I think it makes sense to have the lines render below everything but fills :)